### PR TITLE
API cleanup regarding `Get()` returning interfaces

### PR DESCRIPTION
**Pull request by aricart**
[CHANGE] auth.Operators().Get(name) now returns (Operator,bool) - this bypasses annoying go returning non-nil when interface.

### CODE REVIEW

**Comment by aricart** [comment]
@ripienaar this is a PITA change for you, but I think it is the right thing to do by the API - anywhere we return an interface from a Get/GetByName and the likes, a boolean argument will tell you if the get succeeded. Otherwise checks with nil require reflection, I think this is friendlier API. Some Gets remain as they return concrete types, for those the nil checks are safe in user code.

**Comment by aricart** [comment]
@ripienaar changed so that the bool arguments become error if it is not found. Also added API for knowing if a signing key is present and whether it is scoped.

**Review by ripienaar — CHANGES_REQUESTED** [review]
small consistency notes

File: account_signingkeys.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ func (as *accountSigningKeys) ListRoles() []string {
 	return roles
 }
 
+func (as *accountSigningKeys) Contains(sk string) (bool, bool) {
```
[the remark this answers is not in the document]
Comment (aricart) [reply account_signingkeys.go:47]: one is has the signing key, the other that signing key is scoped.

File: service_imports.go [hunk 2]
```diff
@@ -39,20 +39,24 @@ func (s *serviceImports) AddWithConfig(i ServiceImport) error {
 	return s.update()
 }
 
-func (s *serviceImports) Get(subject string) ServiceImport {
-	return s.getServiceImport(subject)
+func (s *serviceImports) Get(subject string) (ServiceImport, bool) {
```
[the remark this answers is not in the document]
Comment (ripienaar) [reply service_imports.go:42]: Still got this one and similar one below @aricart

File: service_imports.go [hunk 3]
```diff
@@ -39,20 +39,24 @@ func (s *serviceImports) AddWithConfig(i ServiceImport) error {
...
+	si := s.getServiceImport(subject)
+	if si != nil {
+		return si, true
+	}
+	return nil, false
 }
 
-func (s *serviceImports) GetByName(name string) ServiceImport {
+func (s *serviceImports) GetByName(name string) (ServiceImport, bool) {
```
[the remark this answers is not in the document]
Comment (aricart) [reply service_imports.go:50]: oops

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**